File: CifraCesar/CifraCesar.java
import java.util.Scanner;

public class CifraCesar{
   
    public static String encriptar(String texto) {
        StringBuilder textoCifrado = new StringBuilder();
        int n = texto.length(); //Identifica o tamanho da string
        int chave = 3; // determina o valor do deslocamento das letras
        char ch;
            
        for (int i=0; i < n; i++){
            
            if (Character.isUpperCase(texto.charAt(i))){ //Condição para caso letra seja maiúscula
                ch = (char)(((int)texto.charAt(i) + chave - 65) % 26 + 65); 
                textoCifrado.append(ch);
            } 
            else if(texto.charAt(i) == ' '){ //condição para quando um espaço seja encontrado
                ch = ' ';
                textoCifrado.append(ch);
            }
            else {
                ch = (char)(((int)texto.charAt(i) + chave - 97) % 26 + 97); //condição para letra minúscula. 
                textoCifrado.append(ch);
            }
        }
        return textoCifrado.toString();
    }
   
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        String fraseNormal = new String();
        String fraseCifrada = new String();
        
        fraseNormal = sc.nextLine();
        
        while(fraseNormal.equals("FIM") != true){ //Loop é interrompido quando a string FIM é identificada.
            fraseCifrada = encriptar(fraseNormal);
            System.out.printf(fraseCifrada + "\n");
            fraseNormal = sc.nextLine();
        }

        sc.close();
    }
}

File: CifraCesar/CifraCesar.c
#include <stdio.h>
#include <string.h>

void encriptar(char[], char[]);

int main(){
    char fraseNormal[128]; 
    char fraseCifrada[128];

    scanf("%[^\n]s",fraseNormal);
    scanf("%*c");

    while(strcmp(fraseNormal,"FIM") != 0){ //loop interrompe ao identificar a string FIM
        encriptar(fraseNormal,fraseCifrada);
        printf("%s\n", fraseCifrada);
        scanf("%[^\n]s",fraseNormal);
        scanf("%*c");
    }

    return 0;
}

void encriptar(char fraseNormal[128], char fraseEncriptada[128]){
    int chave = 3; //valor da cibra
    int contador = 0;

    for(int i =0; fraseNormal[i] != '\0'; i++){ ///Loop executa até encontrar o '\0'
        
        if(fraseNormal[i] < 91 && fraseNormal[i] > 64){ //Condição para letras maisuculas
            fraseEncriptada[i] = (fraseNormal[i] + chave - 65) % 26 + 65;
            contador++;
        }
        else if(fraseNormal[i] == ' '){ //condição para espaço
            fraseEncriptada[i] = ' ';
            contador++;
        }
        else { //condição para letras minusculas
            fraseEncriptada[i] = (fraseNormal[i] + chave - 97) % 26 + 97;
            contador++;
        }
    }

    fraseEncriptada[contador] = '\0'; //adiciona o '\0' ao final de fraseEncriptada.
}